### ISSUE TITLE
[Android] Exception when adding `MenuItems` for the `NavigationView` while having `PaneDisplayMode="Top"`

### ISSUE THREAD

**Issue by agneszitte** [unoplatform/uno#3856]
## Current behavior 
Java.Lang.IllegalStateException when adding MenuItems for the NavigationView while having PaneDisplayMode="Top" at the same time on Android.

```
Java.Lang.IllegalStateException: 'View with id -1: crc64a5a37c43dff01024.Grid#onMeasure() did not set the measured dimension by calling setMeasuredDimension()'
```
![image](https://user-images.githubusercontent.com/16295702/90714725-4c291300-e276-11ea-80e7-31620360b8d4.png)

## Expected behavior

No exceptions on Android when adding MenuItems for the NavigationView while having PaneDisplayMode="Top" at the same time.

## How to reproduce it (as minimally and precisely as possible)
XAML
```
<NavigationView x:Name="MainNav"
			     PaneDisplayMode="Top" />
```
Code Behind
```
MainNav.MenuItems.Add(new NavigationViewItem()
				{
					Content = page.Title,
					Icon = page.Icon,
					Visibility = page.Visibility,
				});
```

## Environment

Nuget Package: Uno.UI

Package Version(s): 3.1.0-dev.54

Affected platform(s):

- [ ] iOS
- [X] Android
- [ ] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

Visual Studio:

- [ ] 2017 (version: )
- [X] 2019 (version: 16.5.4)
- [ ] for Mac (version: )

## Anything else we need to know?

**Related issue #406**

**Comment by agneszitte-nventive** [comment]
Issue can be tested with this [solution ](https://github.com/yoshiask/AggieMove)at this [commit](https://github.com/yoshiask/AggieMove/commit/5bd173f8d42cb671ad6b71898172b57e1e4e5acc).
You need to uncomment [this part](https://github.com/yoshiask/AggieMove/blob/0f7fb74c4b3cba12427c075036c76655e8faa929/AggieMove/AggieMove.Shared/MainPage.xaml.cs#L34) to be able to replicate the issue.

**Comment by workgroupengineering** [comment]
1. Run SamplesApp.Droid
2. Go to WinUI -> NavigationViewAnimationPage
3. Click on Humburger Menu to open menu
4. Click on Humburger Menu to close menu
5. Click on Flip Orentation button

**Comment by MartinZikmund** [comment]
Happens for WinUI `NavigationView` as well

**Comment by agneszitte-nventive** [comment]
> With version: 3.11.0-dev.487 still occures. Will it be any milestone for it?

cc @jeromelaban

**Comment by morning4coffe-dev** [comment]
@agneszitte @MartinZikmund Was this Issue also reproduceable on blank app? I tested on latest `Uno.Sdk 5.3.0-dev.1827` on Android with Blank app and I wasn't able to repro (similar code as How to reproduce it, with for loop):
![image](https://github.com/unoplatform/uno/assets/78549750/52b7ea3f-b982-4594-bd2c-195229520a0c)

Just a note, the "Related issue" is closed as fixed https://github.com/unoplatform/uno/issues/406

**Comment by agneszitte** [comment]
> @agneszitte @MartinZikmund Was this Issue also reproduceable on blank app? I tested on latest `Uno.Sdk 5.3.0-dev.1827` on Android with Blank app and I wasn't able to repro (similar code as How to reproduce it, with for loop): 
> 
> Just a note, the "Related issue" is closed as fixed #406

@morning4coffe-dev it is an old issue, I think you are the first one giving a try with the latest packages and templates.
@jeromelaban, @MartinZikmund if @morning4coffe-dev is not able to reproduce it, then we can close this issue, right?
